src/core/api/border_router_api.cpp

```diff
@@ -80,18 +80,18 @@ otError otBorderRouterAddOnMeshPrefix(otInstance *aInstance, const otBorderRoute
 
     OT_ASSERT(aConfig != nullptr);
 
-    SuccessOrExit(error = instance.Get<NetworkData::Local>().AddOnMeshPrefix(*config));
+    SuccessOrExit(error = instance.Get<NetworkData::Local>().ValidateOnMeshPrefix(*config));
```

Comment:
I would suggest exactly the same and the method is precisely `AddOnMeshPrefix`. I think we should move `Domain Prefix` specific logic to `AddOnMeshPrefix`, instead of the current strategy of calling `AddOnMeshPrefix` in `SetDomainPrefix. 

That being said, I am fine to enhance it in a separate PR since this is not an issue that introduced by this PR.